[WIP] redesign the whole frontend application

Rintel opened the pull request:
New frontend design! This branch basically touches everything.

This is (as labeled) pretty much work in progress.  We have a lot of dependencies because of these changes. Don't merge!

Commit:
redesign create project template

Commit:
redesign create project template

mharasic committed:
Further changes in style

mharasic committed:
Further changes in style

Drakulix commented on web/src/components/common/SideBar.jsx:
```diff
@@ -17,8 +17,9 @@ export default class SideBar extends Component {
+  //has to be updated if history is working
   render() {
-    let isActive = (url) => { url === this.props.location };
+    let isActive = (url) => false;//{ url === location };
```
why?

mharasic replied:
Because of an awesome bug in locations... now returned

mharasic committed:
minor changes

mharasic committed:
minor changes

mharasic committed:
Edited Project view to match editor style

mharasic committed:
Edited Project view to match editor style

mharasic commented:
Still work in progress as it has to be connected to the backend

mharasic committed:
Color adopted to the theme

mharasic committed:
Color adopted to the theme

mharasic committed:
Further changes

mharasic committed:
Further changes

mharasic committed:
Improved choice how to create project

mharasic committed:
Improved choice how to create project

mharasic committed:
Improved choice how to create project

mharasic committed:
Improved choice how to create project

mharasic committed:
Notification Topbar fixed

mharasic committed:
Notification Topbar fixed

mharasic committed:
Notification Topbar fixed

mharasic committed:
Notification Topbar fixed

mharasic committed:
Minor style improvements

mharasic committed:
Minor style improvements

mharasic committed:
Minor changes

mharasic committed:
Minor changes

mharasic committed:
Minor changes

mharasic committed:
Minor changes

Rintel committed:
Merge branch 'master' into frontend/feature/create-project-redesign

Rintel committed:
rebased to master and fixed dependency conflicts

Rintel committed:
tbc

Rintel committed:
removed react-native package

mharasic committed:
Notifaction pane

mharasic committed:
Notifaction pane

mharasic committed:
Bug fix in Sidebar

mharasic committed:
Minor design changes

mharasic committed:
Sidebar for comments

mharasic committed:
Added missing files

Phil-Schmidt committed:
Bugfix for the file import

-Alert message will appear when an invalid file is read
-Fixed the import for Arrays

Rintel committed:
removed unnecessary tabs in profile view

mharasic committed:
Bug Fixing

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

mharasic committed:
Bug Fix

Rintel committed:
fixed topbar icon positioning

Rintel committed:
resolved merge conflicts

Rintel committed:
reverted fail merge

Rintel committed:
added status badges and changed a link

Phil-Schmidt committed:
Implemented basic error handling for create by link

-Create by link will no longer redirect to crete by form if the provided link does not contain a valid json file and will display an error message.

JamesComey committed:
Frontend

Rintel committed:
Merge pull request #278 from Drakulix/FE/improvements

Frontend

Rintel committed:
fixed merge conflicts

mharasic committed:
XXX

mharasic committed:
XXX

Rintel committed:
removed profile comment tab and fixed a double import

Rintel committed:
added roboto fonts

mharasic committed:
Bug Fixes

mharasic committed:
Bug Fixes

mharasic committed:
Bug Fixes

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Rintel committed:
some changes due to rebase

Rintel committed:
resolved merge conflicts

Rintel committed:
stared with styling of new data table

Rintel committed:
added tabbar in profile menu

JamesComey committed:
Project view funtional and many bugfixes ... Still needs lots of improvements!

mharasic committed:
Filter redesign; Result table as component

mharasic committed:
Filter redesign; Result table as component

mharasic committed:
MAjor Bugfixes

mharasic committed:
Minor Bugfix

mharasic committed:
Minor

mharasic committed:
minor design changes

mharasic committed:
Refactor

mharasic committed:
Cleanup stuff

mharasic committed:
minor changes

mharasic committed:
Prepared for comments

JamesComey committed:
Profileview new tab

JamesComey committed:
Registration and some other minor fixes

mharasic committed:
Updated Table View

mharasic committed:
Improved Table

mharasic committed:
Minor changes

mharasic committed:
Minor changes

gitmirgut committed:
change favicon

mharasic committed:
Finished Fancy DATA Table

mharasic committed:
refactor

mharasic committed:
prepared for bookmarks

JamesComey committed:
Signup and Signin fixes

mharasic committed:
refactor

mharasic committed:
bug fix

mharasic committed:
Minor changes

Rintel committed:
Merge branch 'master' into frontend/feature/create-project-redesign

Phil-Schmidt committed:
JSON5 fileuplaod enabled

- We are now able to add projects by providing links to JSON5 files

mharasic committed:
Minor changes

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

gitmirgut committed:
Remove is_permitted check in get_user from backend api.

JamesComey committed:
Small fic in Signin and project with datatable added in profileview

gitmirgut committed:
Style of DataTable.

gitmirgut committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Phil-Schmidt committed:
Massive merge changes to 'unbreak' online json import

- Merged frontend/feature/create-project-redisign into local branch to enable json5 parsing
- Resolved critical errors in json upload
- restructured the process for project creation by url

Rintel committed:
Merge branch 'master' into frontend/feature/create-project-redesign

mharasic committed:
removed dependencies

mharasic committed:
removed dependencies

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

mharasic committed:
prepared project upload

mharasic committed:
Used backend

mharasic committed:
XXX

mharasic committed:
New Styling

Phil-Schmidt committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/json5-support

Phil-Schmidt committed:
Merge pull request #293 from Drakulix/frontend/feature/json5-support

[Frontend] json5 support

PSDavidSt committed:
minor design changes

moved some things in search and create project to be on same height

pdler committed:
Added admin area funtionalites

pdler committed:
datatable contains real data now

pdler committed:
edit profile overview

mharasic committed:
Minor bug fix for filter

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

pdler committed:
added project delete and bookmarking support

Rintel committed:
Update package.json

Rintel committed:
fixed an infinite alert loop

gitmirgut committed:
update package.json

gitmirgut committed:
update package json

pdler committed:
d

pdler committed:
d

JamesComey committed:
Projects get fetched, fixed some bug where one funtion was suddenly missing?! Still needs improvements when updating profile as authors, date and status get not loaded correctly, I tried but failed to fix this...

pdler committed:
bookmarks works now

pdler committed:
merge

Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

pdler committed:
bookmark star is black if project is bookmarked

pdler committed:
admin area is visible if user is admin

pdler committed:
removed alert messages

pdler committed:
removed more alert messages

gitmirgut committed:
connect comments.

mharasic committed:
Bug fix

gitmirgut committed:
revert to old package.json and add missing comma.

pdler committed:
fixed data in bookmark table

pdler committed:
merge

Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

gitmirgut committed:
Merge branch 'master' into frontend/feature/create-project-redesign

gitmirgut committed:
remove merge conflicts -.-

Rintel committed:
added autocompletion for tags and authors

Rintel committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Rintel committed:
Merge branch 'master' into frontend/feature/create-project-redesign

mharasic committed:
Saved Search

mharasic committed:
Saved search

mharasic committed:
saved search

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Conflicts:
	web/src/components/common/Filters.jsx

gitmirgut committed:
correct ProfileContainer

gitmirgut committed:
revert last commit

mharasic committed:
Saved Query!!! Linking to Backend has to be done

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

mharasic committed:
Small minor issue

mharasic committed:
Tags for Users added

mharasic committed:
double div

mharasic committed:
missing impor

mharasic committed:
Added commentField

gitmirgut committed:
fix backend

gitmirgut committed:
hotfix ProfileView

mharasic committed:
Removed unneccesary imports

mharasic committed:
Merge

pdler committed:
fix bookmark issue in discovery

mharasic committed:
commit bugfix

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Conflicts:
	web/src/components/common/DataTable.jsx
	web/src/components/views/BookmarksTable.jsx

pdler committed:
fix bookmark and own prject view in profile

pdler committed:
merge
x
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

pdler committed:
fixed advanced search in user projects

mharasic committed:
added dynamic bookmarking

pdler committed:
fixed bookmark advanced search

mharasic committed:
Added Dynamic Bookmarking

mharasic committed:
Sharing functionality

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

mharasic committed:
bug fix

pdler committed:
bookmarking is dynamic now

Rintel committed:
added autocompletion for authors in sharepane

gitmirgut committed:
fix simple search.

mharasic committed:
Notification prepared

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

pdler committed:
fix simple search

pdler committed:
merge

Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Rintel committed:
display email instead of user name in comments

mharasic committed:
Notfication styled

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Rintel committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Rintel committed:
added autocompletion for tags and authors in create project by pattern

gitmirgut committed:
fix post of projects.

mharasic committed:
Minor change

mharasic committed:
minor

mharasic committed:
Merge branch 'frontend/feature/create-project-redesign' of https://github.com/Drakulix/knex into frontend/feature/create-project-redesign

Drakulix commented:
:hankey:

Drakulix reviewed (dismissed):
:hankey: :gun:

Drakulix committed:
Merge!
